torch/csrc/distributed/c10d/Ops.cpp

```diff
@@ -515,6 +516,7 @@ REGISTER_C10D_OP(alltoall_)
 REGISTER_C10D_OP(alltoall_base_)
 REGISTER_C10D_OP(barrier)
 
+REGISTER_C10D_OP1(allreduce_, XPU)
```

Comment:
I think we don't want to change this file for each new collective implementation. Then could we register all ops and some ops with unsupported check? Eventually, all collectives should be registered on XPU.